src/Controller/DashboardController.php

```diff
@@ -93,4 +105,35 @@ public function ajaxTab(UserInterface $user, $subpage) {
     return $response;
   }
 
+  /**
+   * Return the AJAX command for changing tab.
+   *
+   * @param string $tab
+   *   The id of the tab being updated.
+   * @param string $block
+   *   The id of the block being moved.
+   * @param string $region
+   *   The region the block is being moved to.
+   * @param int $weight
+   *   The weight of the block in that region.
+   *
+   * @todo return a http response code.
+   * @return array
+   *   The response commands.
+   */
+  public function moveBlock($tab, $block, $region, $weight) {
+    /* @var \Drupal\contacts\Entity\ContactTab $tab */
+    $tab = $this->entityTypeManager()->getStorage('contact_tab')->load($tab);
+
+    $block_config = $tab->getBlock($block);
+    $block_config['region'] = $region;
+    $block_config['weight'] = $weight;
+
+    $tab->setBlock($block, $block_config);
+
+    $tab->save();
+
+    return [];
```

Comment:
I wonder if we want to return the new block/weight or simply a success so the JS can revert changes that fail?